### PR TITLE
added pagination for home, profile, and search

### PR DESCRIPTION
**Pull request by adrean-hipe**
### OVERVIEW

- added pagination for home, profile, and search controllers
- removed unused index method in PostController
- added pagination links to the pages

### SNIPPETS

home pagination
image
profile paginationi - 2nd page
image
search paginnation
image

### CODE REVIEW

File: laravel-app/app/Http/Controllers/ProfileController.php [hunk 1]
```diff
@@ -74,6 +80,7 @@ public function show(int $userId): RedirectResponse|View
         return view('profile.show', [
             'user' => $user,
             'likesCount' => $likesCount,
```
Comment (bpocrafael) [inline laravel-app/app/Http/Controllers/ProfileController.php:82]: just wondering why do you still need this likesCount if you already are adding it to the posts using with.

Comment (adrean-hipe) [reply laravel-app/app/Http/Controllers/ProfileController.php:82]: i added likesCount here sir so I will not have to sum the likes at the show blade

Comment (bpocrafael) [reply laravel-app/app/Http/Controllers/ProfileController.php:82]: I see. okay so the likecounts the total overall of the likes that the user received. the the post is the just for the specific post

Comment (adrean-hipe) [reply laravel-app/app/Http/Controllers/ProfileController.php:82]: yes sir 🙇

Comment (bpocrafael) [reply laravel-app/app/Http/Controllers/ProfileController.php:82]: I see okay then maybe that function of getting the sum of a user you can put it in the model of the user??

Comment (bpocrafael) [reply laravel-app/app/Http/Controllers/ProfileController.php:82]: always remember the concept of fat model , thin controller

Comment (adrean-hipe) [reply laravel-app/app/Http/Controllers/ProfileController.php:82]: definitely sir, added accessor to the user model thank you ❤️

**Review by bpocrafael — APPROVED** [review]
LGTM 🚀

**Pull request closed without merging**